src/nunit-gui/Presenters/XmlPresenter.cs

```diff
@@ -99,5 +99,31 @@ private void DisplayXml()
                 _view.Header = _selectedItem.Name;
             }
         }
+
+        private XmlNode GetFullXml(TestNode testNode)
+        {
+            ResultNode resultNode = _model.GetResultForTest(testNode);
+            XmlNode currentXml;
+            if (resultNode != null)
+            {
+                currentXml = resultNode.Xml.Clone();
+            }
+            else
+            {
+                currentXml = testNode.Xml.Clone();
+                while (currentXml.ChildNodes.Count > 0)
+                {
+                    currentXml.RemoveChild(currentXml.ChildNodes[0]);
```

Comment:
Here, I believe you are removing child nodes that should not be removed: properties for example, possibly element content. I think you need to have a method that removes child tests only, that is, elements with the name "test-case" or "test-suite".